src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -862,12 +943,50 @@ static ucs_status_t uct_tcp_sockcm_ep_server_create(uct_tcp_sockcm_ep_t *tcp_ep,
               tcp_ep->fd, params_tcp_sockcm, tcp_ep->state);
 
     /* now that the server's ep was created, can try to send data */
-    ucs_async_modify_handler(tcp_ep->fd, UCS_EVENT_SET_EVWRITE | UCS_EVENT_SET_EVREAD);
-    return UCS_OK;
+    ucs_async_modify_handler(tcp_ep->fd, UCS_EVENT_SET_EVWRITE |
+                                         UCS_EVENT_SET_EVREAD);
+
+    if (ucs_test_all_flags(params->field_mask,
+                           UCT_EP_PARAM_FIELD_PRIV_DATA |
+                           UCT_EP_PARAM_FIELD_PRIV_DATA_LENGTH)) {
+        conn_params.field_mask = UCT_EP_CONNECT_PARAM_FIELD_PRIVATE_DATA |
+                                 UCT_EP_CONNECT_PARAM_FIELD_PRIVATE_DATA_LENGTH;
+        conn_params.private_data        = params->private_data;
+        conn_params.private_data_length = params->private_data_length;
+    } else if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) {
+        status = ucs_sockaddr_get_ifname(tcp_ep->fd, ifname_str,
+                                         sizeof(ifname_str));
+        if (UCS_OK != status) {
+            goto err_unblock;
+        }
+
+        pack_args.field_mask = UCT_CM_EP_PRIV_DATA_PACK_ARGS_FIELD_DEVICE_NAME;
+        ucs_strncpy_safe(pack_args.dev_name, ifname_str, UCT_DEVICE_NAME_MAX);
+        data_buf = ucs_malloc(tcp_sockcm->priv_data_len, "tcp_priv_data");
+        status   = uct_cm_ep_pack_cb(&tcp_ep->super, tcp_ep->super.user_data,
+                                     &pack_args, data_buf,
+                                     tcp_sockcm->priv_data_len,
+                                     &priv_data_ret);
+        if (status != UCS_OK) {
+            tcp_ep->state |= UCT_TCP_SOCKCM_EP_PACK_CB_FAILED;
+            goto err_unblock;
+        }
+
+        conn_params.field_mask = UCT_EP_CONNECT_PARAM_FIELD_PRIVATE_DATA |
+                                 UCT_EP_CONNECT_PARAM_FIELD_PRIVATE_DATA_LENGTH;
+        conn_params.private_data        = data_buf;
+        conn_params.private_data_length = priv_data_ret;
+    } else {
+        conn_params.field_mask = 0;
+    }
+
+    status = uct_tcp_sockcm_ep_connect(&tcp_ep->super.super.super,
```

Comment:
in tcp_sockcm, as opposed to rdmacm, ep_connect is called only from the server side? (even though it's an api function to be called from the client side?)

Reply:
should be resolved the same way as https://github.com/openucx/ucx/pull/6503#discussion_r598261158 for consistency